### PR TITLE
feat(exec)!: run exec over the WebSocket bridge with durable sessions

### PR DESCRIPTION
**Pull request by lawrencegripper**
## What

`exec` now runs entirely over the sandbox WebSocket bridge (`/ws/exec`). This
replaces the previous GraphQL exec path, giving separated stdout/stderr, a real
exit code, durable sessions you can reconnect to, and signal-based termination.

## API

```ts
const handle = sandbox.exec("npm run build", { timeoutSec: 120 });
const result = await handle; // { exitCode, stdout, stderr, truncated, timedOut }
```

- Awaiting the handle yields the final `ExecResult`. A non-zero exit does **not**
  throw — inspect `exitCode`.
- `onStdout` / `onStderr` stream output live (stdout and stderr are separate).
- `handle.sessionName` — the durable session id. Reconnect from anywhere with
  `sandbox.exec({ sessionName })`. It rejects if the server assigned no durable
  session (durable sessions unavailable), while the command still runs.
- `handle.detach()` — stop streaming and close the socket **without** ending the
  command; resolves the `sessionName` to reconnect with.
- `handle.kill(signal = "TERM")` — terminate the command's process group with a
  real signal (`"TERM"` | `"KILL"` | `"INT"` | `"QUIT"` | `"HUP"`).
- `ExecOptions.resumeFromLastRead` (default `false`): on reconnect, replay the
  full retained log (lossless, may repeat output an earlier reader saw). Set
  `true` to resume from the server's last-read cursor instead (exact, but can
  drop in-flight output at an abrupt disconnect).
- `timeoutSec` — client-side deadline; resolves with `timedOut: true`.

## Transport & config

- Connects to `wss://ssh.<host>:2226/ws/exec`, derived from the GraphQL endpoint;
  override with `tcpProxyWsEndpoint` / `RAILWAY_TCP_PROXY_WS_ENDPOINT`.
- Authorized by a short-lived shell token.
- Requires a global `WebSocket` (Node 22+, browsers, edge) or pass `webSocketImpl`.

## Breaking changes

- exec is WebSocket-only: the GraphQL exec operations and the `transport` option
  are removed.
- `ExecHandle.execId` → `ExecHandle.sessionName` (and `exec({ execId })` →
  `exec({ sessionName })`).
- `kill()` takes an optional signal and now sends a real signal (previously it
  closed the socket).

## Notes

- Reattaching to an already-expired session can return an empty successful result
  rather than an error (the bridge starts a fresh session for an unknown id).

## Tests

- Offline unit suite against a fake WebSocket — no network.
- Live e2e (gated by `SANDBOX_E2E`): split streams, live streaming, `timeoutSec`,
  durable reattach, `detach()`, and `kill()` — verified against a durable-enabled
  bridge.

### CODE REVIEW

**Comment by lawrencegripper** [comment]
image

**Review by coffee-cup — APPROVED** [review]

**Review by coffee-cup — APPROVED** [review]
